File: backoffice/src/diningTable/domain/valueObjects/DiningTableStatus.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain import IntValueObject

"""
 *
 * Classes 
 *
"""

class DiningTableStatus( IntValueObject ):

    """
     *
     * Constants 
     *
    """

    __ENABLED  = 1
    __DELETED  = 2

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : int ) -> None:
        super().__init__( value )
    
    @classmethod
    def createEnabled( cls ) -> object:
        return cls( cls.__ENABLED )
    
    @classmethod
    def createDeleted( cls ) -> object:
        return cls( cls.__DELETED )

File: preparation/src/diningTable/domain/valueObjects/DiningTableStatus.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class DiningTableStatus : IntValueObject {

        /* 
         * 
         * Constants
         *
        */

        private const int ENABLED = 1;
        private const int DELETED = 2;

        /* 
         * 
         * Methods
         *
        */

        public DiningTableStatus( int value ) : base( value ) {
        }

        public static DiningTableStatus createEnabled() {
            return new DiningTableStatus( ENABLED );
        }

        public static DiningTableStatus createDeleted() {
            return new DiningTableStatus( DELETED );
        }

    }

}